src/test/blockchain/blockchain_parameters_tests.cpp

```diff
@@ -142,4 +143,23 @@ BOOST_AUTO_TEST_CASE(difficulty_function_max_test) {
   }
 }
 
+void CheckGenesisBlockSupply(const blockchain::Parameters &p) noexcept {
+  CAmount genesis_supply = 0;
+  for (const auto &tx : p.genesis_block.block.vtx) {
+    genesis_supply += tx->GetValueOut();
+  }
+  BOOST_CHECK_EQUAL(genesis_supply, p.initial_supply);
+}
+
+BOOST_AUTO_TEST_CASE(genesis_supply_test) {
+  const auto base_parameters = blockchain::Parameters::Base();
+  CheckGenesisBlockSupply(base_parameters);
+
+  const auto testnet_parameters = blockchain::Parameters::TestNet();
+  CheckGenesisBlockSupply(testnet_parameters);
```

Comment:
why not have that as an assertion in the blockchain parameters?

Reply:
It was like that three commits ago :sweat_smile: . Pros and cons? For me it was more natural that way, but it also adds more overhead to the binary.

Reply:
>  it also adds more overhead to the binary

So might `-O3`. If you want a slim binary, do `-Os`.